src/main/java/com/endava/internship/collections/StudentMap.java

```diff
@@ -1,72 +1,396 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Consumer;
 
 public class StudentMap implements Map<Student, Integer> {
+
+    private final int DEFAULT_CAPACITY = 1 << 4;
+    private final int MAXIMUM_CAPACITY = 1 << 30;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    final class Node {
+        final int hash;
+        final Student key;
+        Integer value;
+        Node next;
+
+        Node(int hash, Student key, Integer value, Node next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+        public final Student getKey() {
+            return key;
+        }
+        public final Integer getVal() {
+            return value;
+        }
+        public final int getHash() {
+            return hash;
+        }
+        public final Node getNext() {
+            return next;
+        }
+        public void setVal(Integer value) {
+            this.value = value;
+        }
+        public void setNext(Node next) {
+            this.next = next;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if(obj instanceof Node) {
+                Node another = ((Node) obj);
+                return this.getHash() == another.getHash() &&                   // если равны по hash коду
```

Comment:
Hash is not the parameter that we need to look when comparing equals. There is a small posibility that two different objects could have same hash, please impliment this the same way you used for the student equals

Reply:
I agree, so after checking the hash, I compare the values with the equals method. But if the objects are not even in the hash code, there is no point in comparing them using the equals method.

Reply:
I do understand this logic, but I simply don't like it, cause hashcode can be sometime overriden incorectly. so equals method not must apply on hashcode

Reply:
btw, node is equal if the key & value are equals

Reply:
Good, fixed